torch/_dynamo/variables/builtin.py

```diff
@@ -998,9 +1002,10 @@ def call_super(self, tx, a, b):
         return variables.SuperVariable(a, b)
 
     def call_next(self, tx, arg):
-        if isinstance(arg, variables.ListIteratorVariable):
-            val, next_iter = arg.next_variables()
-            tx.replace_all(arg, next_iter)
+        if isinstance(
+            arg, (variables.ListIteratorVariable, variables.IteratorVariable)
+        ):
+            val, next_iter = arg.next_variables(tx)
```

Comment:
oh ok, no bug, you just moved where the replacement happens. You sure you don't want the replacement to happen here?

Reply:
as mentioned, we need to move the tx into the IteratorVariable as some of the logic in `next_variables` need to ref it